### ISSUE TITLE
[DataGridPro] - Lazy loading with a large data-set (> 1M rows) is not usable

### ISSUE THREAD

**Issue by formatlos** [mui/mui-x#8085]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

Steps:
1. Use lazyLoading like described here https://mui.com/x/react-data-grid/row-updates/#lazy-loading just with 10000000 rows
2. Start scrolling 

### Current behavior 😯

As soon as a slice of data is loaded and added to the grid the browser freezes. The grid gets very slow and un-reactive overall. 

From quickly looking into the source: you seem to loop over `rowCount` in `useGridLazyLoaderPreProcessors` to initialize skeleton rows. 

### Expected behavior 🤔

I would expect the Data Grid is usable and fast no matter how large the result list is. That's obviously the reason for using lazy loading and virtualization  in the first place. 

### Context 🔦

Displaying a Data Grid with more than 1M rows.

### Your environment 🌎

  System:
    OS: macOS 13.1
  Binaries:
    Node: 16.13.0
    npm: 8.8.0
  Browsers:
    Chrome: 110.0.5481.177
    Edge: Not Found
    Firefox: 109.0.1
    Safari: 16.2

### Order ID 💳 (optional)

58497

**Comment by formatlos** [comment]
Yes I know, but that makes the grid unusable for datasources with large or unknown size, which is basically the main use-case of lazy-loading in the first place. 

Is there some possibility to opt out from creating fake rows?

**Comment by formatlos** [comment]
> > Is there some possibility to opt out from creating fake rows?
> 
> No, we'll to improve this part to make it usable with large datasets.

@m4theushw sounds good, do you have a schedule for this?

**Comment by joserodolfofreitas** [comment]
In an internal discussion, we converged on making the skeleton rows lightweight by not loading them on the internal state.

> do you have a schedule for this?

Not concretely, but it's on the short-medium term horizon.

**Comment by BrentFarese** [comment]
Is there really no solution to this yet?  For lazy-loading use cases, the DataGrid is practically unusable w/o this?  We have a data set with 10K rows and it is unusable/the browser freezes due to this issue.  Is there a timeline on a fix?  We are DataGridPremium subscribers.

This is sort of unbelievable, the entire point of lazy-loading is to only fetch X rows and for it to be performant (and for it to handle decent size data sets).  Why load all skeleton rows?  There are other ways to do it and it's likely not a big deal if the scroll bar changes height during loading (in contrast it's a huge deal that the browser freezes up and it's basically unusable).

Please fix.  Thank you!
